ProcessLib/UncoupledProcessesTimeLoop.h

```diff
@@ -29,24 +34,44 @@ class UncoupledProcessesTimeLoop
     explicit UncoupledProcessesTimeLoop(
         std::unique_ptr<NumLib::ITimeStepAlgorithm>&& timestepper,
         std::unique_ptr<Output>&& output,
-        std::vector<std::unique_ptr<SingleProcessData>>&& per_process_data);
+        std::vector<std::unique_ptr<SingleProcessData>>&& per_process_data,
+        const unsigned global_coupling_max_iterations,
+        std::unique_ptr<NumLib::ConvergenceCriterion>&& glb_coupling_conv_crit);
 
     bool loop();
 
     ~UncoupledProcessesTimeLoop();
 
+    bool setCoupledSolutions();
+
 private:
     std::vector<GlobalVector*> _process_solutions;
     std::unique_ptr<NumLib::ITimeStepAlgorithm> _timestepper;
     std::unique_ptr<Output> _output;
     std::vector<std::unique_ptr<SingleProcessData>> _per_process_data;
+
+    /// Maximum iterations of the global coupling.
+    const unsigned _global_coupling_max_iterations;
+    /// Convergence criteria of the global coupling iterations.
+    std::unique_ptr<NumLib::ConvergenceCriterion> _global_coupling_conv_crit;
+
+    /**
+     *  Vector of solutions of coupled processes of processes.
+     *  Each vector element stores the references of the solution vectors
+     *  (stored in _process_solutions) of the coupled processes of a process.
+     */
+    std::vector<std::map<ProcessType, GlobalVector const&>>
```

Comment:
Different from what I presented in the meeting, I changed ``std::vector<std::map<ProcessType, GlobalVector const*>>`` to the current more readable one.